acapy_agent/wallet/askar.py

```diff
@@ -131,14 +135,58 @@ async def assign_kid_to_key(self, verkey: str, kid: str) -> KeyInfo:
         if not key_entry:
             raise WalletNotFoundError(f"No key entry found for verkey {verkey}")
 
+        try:
+            existing_kid = key_entry.tags.get("kid")
+        except Exception:
+            existing_kid = []
+
+        existing_kid = existing_kid if isinstance(existing_kid, list) else [existing_kid]
+        existing_kid.append(kid)
+        tags = {"kid": existing_kid}
+
         key = cast(Key, key_entry.key)
         metadata = cast(dict, key_entry.metadata)
         key_types = self.session.inject(KeyTypes)
         key_type = key_types.from_key_type(key.algorithm.value)
         if not key_type:
             raise WalletError(f"Unknown key type {key.algorithm.value}")
 
-        await self._session.handle.update_key(name=verkey, tags={"kid": kid})
+        await self._session.handle.update_key(name=verkey, tags=tags)
+        return KeyInfo(verkey=verkey, metadata=metadata, key_type=key_type, kid=kid)
+
+    async def remove_kid_from_key(self, kid: str) -> KeyInfo:
+        """Remove a kid association.
+
+        Args:
+            kid: the key identifier
+
+        Returns:
+            The key identified by kid
+
+        """
+        key_entries = await self._session.handle.fetch_all_keys(
+            tag_filter={"kid": kid}, limit=2
+        )
+        if len(key_entries) > 1:
+            raise WalletDuplicateError(f"More than one key found by kid {kid}")
+
+        entry = key_entries[0]
+        existing_kid = entry.tags.get("kid")
+        key = cast(Key, entry.key)
+        verkey = bytes_to_b58(key.get_public_bytes())
+        metadata = cast(dict, entry.metadata)
+        key_types = self.session.inject(KeyTypes)
+        key_type = key_types.from_key_type(key.algorithm.value)
+        if not key_type:
+            raise WalletError(f"Unknown key type {key.algorithm.value}")
+        try:
+            existing_kid.remove(kid)
+        except Exception:
```

Comment:
Should catch the correct type of exception for removing entry that doesn't exist. Can't remember of the top of my head.

In general try to only catch the base Exception class when nessecary.